dm/dm.go

```diff
@@ -142,6 +161,10 @@ func execute() {
 
 	switch args[0] {
 	case "templates":
+		if len(args) < 2 {
+			fmt.Fprintln(os.Stderr, "No registry name supplied")
```

Comment:
Agreed. 

We already have a mental for specifying a repo... There's a default and you can set it with the --repository flag. Given that mental model, I don't see a need for a repository argument. We always use the flag.

To go the other way mean either
* being inconsistent from subcommand to subcommand, so that some take arguments and others take the flag, which seems undesirable, or
* always taking the value from the flag, and never using an explicit argument for the repository name.